### ISSUE TITLE
XML of DrudeLangevinIntegrator deserialization returns a DrudeIntegrator object

### ISSUE THREAD

**Issue by AnastasiaCroitoru** [openmm/openmm#4088]
I am using openmm_8.0.0  (py311h59c6c42_1) obtained from  conda-forge.

I noticed that reconstructing a "DrudeLangevinIntegrator" object that has been serialized as XML results in "DrudeIntegrator". The same error is present when trying to `copy.deepcopy()` the integrator.
How to fix it? Is there an issue in my script?

```python
integrator = DrudeLangevinIntegrator(300*kelvin, 5/picosecond, 1*kelvin, 20/picosecond, 0.001*picoseconds)
serialized_integrator = XmlSerializer.serialize(integrator)
print(serialized_integrator)
deserialized_integrator = XmlSerializer.deserialize(serialized_integrator)
print(type(deserialized_integrator))`
```

Output:
```
<?xml version="1.0" ?>
<Integrator constraintTolerance="1e-05" drudeFriction="20" drudeTemperature="1" friction="5" randomSeed="0" stepSize=".001" temperature="300" type="DrudeLangevinIntegrator" version="1"/>

<class 'openmm.openmm.DrudeIntegrator'>
```

**Comment by peastman** [comment]
Thanks!  I see the problem.  When generating the SWIG wrappers, we make a list of every class that extends Integrator (and also Force or TabulatedFunction) so we can build the factories:

https://github.com/openmm/openmm/blob/a6c3641861896e69db584e4e0e87f70d37389252/wrappers/python/src/swig_doxygen/swigInputBuilder.py#L257-L267

It identifies them based on the `basecompoundref` field in the XML file, which it expects to be `OpenMM::Integrator`.  But for DrudeLangevinIntegrator it isn't.  That's because it subclasses DrudeIntegrator, which in turn subclasses Integrator:

```xml
    <basecompoundref refid="classOpenMM_1_1DrudeIntegrator" prot="public" virt="non-virtual">OpenMM::DrudeIntegrator</basecompoundref>
```

We need to make it recognize indirect subclasses as well as direct ones.

**Comment by peastman** [comment]
The fix is in #4089.

**Comment by AnastasiaCroitoru** [comment]
Thanks!
